### PR TITLE
[CARBONDATA-1081] Use binarySearch to replace for clause to improve performance

### PR DESCRIPTION
**Pull request by mayunSaicmotor**
use binarySearch to replace for clause in  method setFilterdIndexToBitSet of ExcludeFilterExecuterImpl.java

### CODE REVIEW

File: core/src/main/java/org/apache/carbondata/core/scan/filter/executer/ExcludeFilterExecuterImpl.java [hunk 1]
```diff
@@ -146,9 +146,17 @@ private BitSet setFilterdIndexToBitSet(FixedLengthDimensionDataChunk dimColumnDa
     BitSet bitSet = new BitSet(numerOfRows);
     bitSet.flip(0, numerOfRows);
     byte[][] filterValues = dimColumnExecuterInfo.getFilterKeys();
-    for (int k = 0; k < filterValues.length; k++) {
+    if (filterValues.length > 1) {
```
Comment (kumarvishal09) [inline core/src/main/java/org/apache/carbondata/core/scan/filter/executer/ExcludeFilterExecuterImpl.java]: This code will not work when No Inverted index is configured for that column while creating the column. In that case data wont be sorted and this code will five wrong result and whether filter value is one or more than one this if data is sorted we can add binary search

**Comment by kumarvishal09** [comment]
retest this please

**Comment by kumarvishal09** [comment]
retest this please

**Comment by kumarvishal09** [comment]
retest this please

**Comment by kumarvishal09** [comment]
retest this please

**Comment by gvramana** [comment]
retest this please

**Pull request closed without merging**